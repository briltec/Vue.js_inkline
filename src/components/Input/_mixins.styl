-form-input-selector()
    > input
    > select
    > textarea
        {block}
