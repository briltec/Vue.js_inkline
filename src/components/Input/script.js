import IFormGroup from 'inkline/components/FormGroup';

import AttributesProviderMixin from 'inkline/mixins/components/providers/AttributesProviderMixin';
import ClassesProviderMixin from 'inkline/mixins/components/providers/ClassesProviderMixin';
import InjectParentFormProviderMixin from 'inkline/mixins/forms/providers/InjectParentFormProviderMixin';
import ModelProviderMixin from 'inkline/mixins/forms/providers/ModelProviderMixin';

import ClickInputRefMethodMixin from 'inkline/mixins/forms/methods/ClickInputRefMethodMixin';
import FocusInputRefMethodMixin from 'inkline/mixins/forms/methods/FocusInputRefMethodMixin';
import EmitChangeMethodMixin from 'inkline/mixins/components/methods/EmitChangeMethodMixin';
import EmitClickMethodMixin from 'inkline/mixins/components/methods/EmitClickMethodMixin';
import EmitFocusMethodMixin from 'inkline/mixins/components/methods/EmitFocusMethodMixin';
import EmitHoverMethodMixin from 'inkline/mixins/components/methods/EmitHoverMethodMixin';
import EmitInputMethodMixin from 'inkline/mixins/components/methods/EmitInputMethodMixin';
import EmitKeydownMethodMixin from 'inkline/mixins/components/methods/EmitKeydownMethodMixin';
import EmitKeyupMethodMixin from 'inkline/mixins/components/methods/EmitKeyupMethodMixin';

import ClearablePropertyMixin from 'inkline/mixins/forms/properties/ClearablePropertyMixin';
import DisabledPropertyMixin from 'inkline/mixins/forms/properties/DisabledPropertyMixin';
import ParentFormGroupPropertyMixin from 'inkline/mixins/forms/properties/ParentFormGroupPropertyMixin';
import ReadonlyPropertyMixin from 'inkline/mixins/forms/properties/ReadonlyPropertyMixin';
import SizePropertyMixin from 'inkline/mixins/components/properties/SizePropertyMixin';
import TabIndexPropertyMixin from 'inkline/mixins/components/properties/TabIndexPropertyMixin';

export default {
    name: 'IInput',
    inheritAttrs: false,
    components: {
        IFormGroup
    },
    mixins: [
        AttributesProviderMixin,
        ClassesProviderMixin,
        InjectParentFormProviderMixin,
        ModelProviderMixin,

        ClickInputRefMethodMixin,
        FocusInputRefMethodMixin,
        EmitChangeMethodMixin,
        EmitClickMethodMixin,
        EmitFocusMethodMixin,
        EmitHoverMethodMixin,
        EmitInputMethodMixin,
        EmitKeydownMethodMixin,
        EmitKeyupMethodMixin,

        ClearablePropertyMixin,
        DisabledPropertyMixin,
        ParentFormGroupPropertyMixin,
        ReadonlyPropertyMixin,
        SizePropertyMixin,
        TabIndexPropertyMixin
    ],
    created () {
        this.classesProvider.add('root', () => ({
            '-prepended': this.$slots.prepend || this.prepended,
            '-appended': this.$slots.append || this.appended
        }));

        this.classesProvider.add('child', () => ({
            '-prefixed': this.$slots.prefix || this.prefixed,
            '-suffixed': this.$slots.suffix || this.suffixed
        }));
    }
};
