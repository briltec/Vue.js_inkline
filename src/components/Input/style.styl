@import '../../resources/css/config'

--label-margin-bottom := 0.5rem

--input-padding := --spacers['1-2'] --spacers['1']
--input-line-height := --line-height

--input-color := --colors['gray-70']
--input-color-focus := --input-color
--input-placeholder-color := --colors['gray-60']
--input-plaintext-color := --body-color
--input-background := --colors['white']
--input-background-disabled := --colors['gray-20']
--input-background-focus := --input-background

--input-border-width := --border-width
--input-border-color := --border-color
--input-border-color-focus := lighten(--input-border-color, 25%)
--input-border-radius := --border-radius

--input-box-shadow := inset 0 1px 1px rgba(--colors['black'], 0.05)
--input-transition := border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out


.form-input
    > input
    > select
    > textarea
        display: block
        width: 100%
        padding: --input-padding
        line-height: --input-line-height
        color: --input-color
        background-color: --input-background
        background-clip: padding-box
        border: --input-border-width solid --input-border-color

        if --border-radius-enabled
            border-radius: --input-border-radius

        box-shadow: --input-box-shadow
        if --transition['input']
            transition: --input-transition

        // Remove style for the caret on `<select>`s in IE10+.
        &::-ms-expand
            background-color: transparent
            border: 0

        // Customize the `:focus` state to imitate native WebKit styles.
        &:focus
            color: --input-color-focus
            background-color: --input-background-focus
            border-color: --input-border-color-focus
            outline: 0;

            // Avoid using mixin so we can pass custom focus shadow properly
            if --box-shadow-enabled
                box-shadow: --input-box-shadow, --input-box-shadow-focus

        // Placeholder
        // Override Firefox's unusual default opacity see https://github.com/twbs/bootstrap/pull/11526.
        &::placeholder
            color: --input-placeholder-color
            opacity: 1

        // Disabled and read-only inputs
        //
        // HTML5 says that controls under a fieldset > legend:first-child won't be
        // disabled if the fieldset is disabled. Due to implementation difficulty, we
        // don't honor that edge case we style them as disabled anyway.
        &:disabled,
        &[readonly]
            background-color: --input-background-disabled
            opacity: 1  // iOS fix for unreadable disabled content

