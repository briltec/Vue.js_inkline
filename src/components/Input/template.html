<div class="form-input"
     @mouseenter="onMouseEnter"
     @mouseleave="onMouseLeave"
     :class="classes">

    <span class="form-input-prefix" v-if="$slots.prefix">
        <slot name="prefix"></slot>
    </span>

    <span class="form-input-suffix" v-if="$slots.suffix || clearable">
        <i class="form-input-icon -circle" aria-label="Clear" v-show="isClearable" @click="clear">&times;</i>
        <slot name="suffix"></slot>
    </span>

    <input
        v-bind="$attrs"
        v-model="model"
        ref="input"
        :type="type"
        :disabled="isDisabled"
        @focus="onFocus"
        @blur="onBlur"
        @change="onChange">
</div>
