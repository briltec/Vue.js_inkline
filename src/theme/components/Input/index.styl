//
// Variables
//
--input-append-prepend-background-gradient := --gradients['gray-30']

//
// Styles
//
.form-input-prepend
.form-input-append
    background-image: -linear-gradient(--input-append-prepend-background-gradient)
