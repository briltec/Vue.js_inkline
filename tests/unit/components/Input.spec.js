import { mount } from '@vue/test-utils';
import Input from 'inkline/components/Input';

describe('Components', () => {
    describe('Input', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = mount(Input);
        });

        test('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
