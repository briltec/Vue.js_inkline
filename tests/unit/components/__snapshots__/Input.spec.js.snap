// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Input should render correctly 1`] = `<div role="checkboxgroup" class="button-group form-group -checkbox"></div>`;
